x-pack/legacy/plugins/task_manager/task_runner.ts

```diff
@@ -166,12 +166,23 @@ export class TaskManagerRunner implements TaskRunner {
    */
   public async claimOwnership(): Promise<boolean> {
     const VERSION_CONFLICT_STATUS = 409;
+    const attempts = this.instance.attempts + 1;
+    const now = new Date(Date.now());
+
+    const timeoutDate = intervalFromNow(this.definition.timeout!)!;
 
     try {
       this.instance = await this.store.update({
         ...this.instance,
         status: 'running',
-        runAt: intervalFromNow(this.definition.timeout)!,
+        startedAt: now,
+        attempts,
+        retryAt: new Date(
+          timeoutDate.getTime() +
+            (this.definition.getRetryDelay
+              ? this.definition.getRetryDelay(attempts, Boom.clientTimeout()) * 1000
```

Comment:
seems slightly weird to create an error just to pass into `getRetryDelay()`, and not sure why a `Boom` error would be needed.  Also seems weird to set the retryAt when the task is being "owned" - I guess that's a safety net?  I'd expect it would be reset based on a real error encountered, when the task is being "put back on the queue".

Reply:
In regards to the Boom error, it was the only way I could think of to call `getRetryDelay` with an error of timeout (not much difference using theirs vs creating our own). I could always pass nothing as error to `getRetryDelay` but if someone wants to do special timeout logic, they can by looking for this given error.

In regards to the `retryAt`, this is used to determine if the claimed owner crashed or something and didn't finish the task on time. This will cause a new owner to claim it and attempt the task again.

Reply:
cool - a comment on why `Boom.clientTimeout()` was used would be useful I think